fdb-relational-core/src/main/java/com/apple/foundationdb/relational/recordlayer/query/Expressions.java

```diff
@@ -101,7 +102,8 @@ public Expressions pullUp(@Nonnull Value value, @Nonnull CorrelationIdentifier c
                                 simplifiedValue.pullUp(List.of(subExpression), EvaluationContext.empty(), aliasMap,
                                         constantAliases, correlationIdentifier);
                         if (pulledUpExpressionMap.containsKey(subExpression)) {
-                            return pulledUpExpressionMap.get(subExpression);
+                            Assert.thatUnchecked(pulledUpExpressionMap.get(subExpression).size() == 1, ErrorCode.AMBIGUOUS_COLUMN, "Ambiguous columns for " + subExpression);
```

Comment:
Same here, I think it is fine to pull some of these up here even though there are ambiguities. This code needs any pulled up version and does not care much which one it gets.